sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -175,6 +175,27 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "The default namespace cannot be deduced in v2 catalog yet")
```

Comment:
This is not true now. If the catalog implements `SupportsNamespaces`, it has a `defaultNamespace` method.

Reply:
This is correct for now because this should use the current namespace, which we don't track for v2 yet.

Reply:
ah i see, we have default namespace but no current namespace.

Reply:
I updated the message to "The current namespace is not available in v2 catalog yet" to avoid confusion.